client/src/main/java/org/apache/rocketmq/client/log/Log4j2Helper.java

```diff
@@ -0,0 +1,159 @@
+import org.apache.logging.log4j.core.appender.AsyncAppender;
+import org.apache.logging.log4j.core.appender.RollingFileAppender;
+import org.apache.logging.log4j.core.appender.rolling.CompositeTriggeringPolicy;
+import org.apache.logging.log4j.core.appender.rolling.DefaultRolloverStrategy;
+import org.apache.logging.log4j.core.appender.rolling.SizeBasedTriggeringPolicy;
+import org.apache.logging.log4j.core.appender.rolling.TimeBasedTriggeringPolicy;
+import org.apache.logging.log4j.core.config.AppenderRef;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.LoggerConfig;
+import org.apache.logging.log4j.core.layout.PatternLayout;
+
+import java.lang.reflect.Method;
+
+public class Log4j2Helper {
+
+    private static final int ASYNC_APPENDER_QUEUE_SIZE = 1024;
+
+    private static final String APPENDER_FILE_SIZE = "100MB";
+
+    private static final int ASYNC_APPENDER_SHUTDOWN_TIMEOUT = 3;
+
+    private static final String LOG_LAYOUT_PATTERN = "%d{yyy-MM-dd HH:mm:ss,GMT+8} %p %t - %m%n";
+
+    private static Method getMethodWithOnlyName(Class clazz, String name) {
+        Method[] declaredMethods = clazz.getDeclaredMethods();
+        for (Method declaredMethod : declaredMethods) {
+            if (declaredMethod.getName().equals(name)) {
+                return declaredMethod;
+            }
+        }
+        return null;
+    }
+
+    private static DefaultRolloverStrategy createRolloverStrategy(String max, final Configuration config) {
+        Method createMethod = getMethodWithOnlyName(DefaultRolloverStrategy.class, "createStrategy");
+        if (createMethod != null) {
+            createMethod.setAccessible(true);
+            Class<?>[] parameterTypes = createMethod.getParameterTypes();
+            try {
+                if (parameterTypes.length == 5) {
+                    return (DefaultRolloverStrategy) createMethod.invoke(null, max, "1", null, "1", config);
+                } else {
+                    return (DefaultRolloverStrategy) createMethod.invoke(null, max, "1", null, "1", null, true, config);
+                }
+            } catch (Exception e) {
+                System.err.println(e);
+            }
+        }
+        return null;
+    }
+
+    private static AbstractAppender createAsyncAppender(AbstractAppender appender, boolean blocking, int queueSize, Configuration config) {
+        String refAppenderName = appender.getName();
+        AppenderRef ref = AppenderRef.createAppenderRef(refAppenderName, null, null);
+        AppenderRef[] refs1 = new AppenderRef[]{ref};
+        String asyncAppenderName = "Async" + refAppenderName;
+
+        Method createMethod = getMethodWithOnlyName(AsyncAppender.class, "createAppender");
+
+        if (createMethod == null) {
+            return appender;
+        }
+        AsyncAppender asyncAppender = null;
+        try {
+            if (createMethod.getParameterTypes().length == 9) {
```

Comment:
Do we have another approach to distinguish these two methods, instead of using magic number 9?